### PR TITLE
[exporter/sumologicexporter]: Removed deprecated compress encoding in sumo exporter

### PR DESCRIPTION
**Pull request by chan-tim-sumo**
**Description**:

Switched to use compression from client config, [compress_encdoing is deprecated](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/fc363988211d90ee2ae7c75eda6ce90c04bae868/exporter/sumologicexporter/config.go#L31), so removing from exporter. 

Link to tracking Issue:
- https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/33138

**Testing**:

Unit test

**Documentation**:

chloggen

### CODE REVIEW

File: exporter/sumologicexporter/sender_test.go
Comment (chan-tim-sumo) [inline exporter/sumologicexporter/sender_test.go:72]: Kept Gzip since this is the default value (safest way?). 

Another way I can think of is:
1. to remove default since switch statement covers all cases: empty string ("") and [other types of compressions](https://pkg.go.dev/go.opentelemetry.io/collector/config/configcompression#Type)

File: .chloggen/chan-tim_deprecateCompressEncoding
Comment (sumo-drosiek) [inline .chloggen/chan-tim_deprecateCompressEncoding:4]: It is `breaking` or `deprecation` change. We should provide in this PR also migration steps for customers

File: exporter/sumologicexporter/README.md
Comment (sumo-drosiek) [inline exporter/sumologicexporter/README.md:83]: we are missing `zstd` here

Comment (sumo-drosiek) [inline exporter/sumologicexporter/README.md:83]: We are missing `zstd` here

File: exporter/sumologicexporter/sender_test.go
Comment (sumo-drosiek) [reply exporter/sumologicexporter/sender_test.go:72]: We would like to avoid it, as it may require migration on customer side. For this option it shouldn't be a big issue, but technically changind default is an another breaking change.

We probably will introduce it (as changing `zstd` to default) in near future 😅

File: .chloggen/chan-tim_deprecateCompressEncoding
Comment (chan-tim-sumo) [reply .chloggen/chan-tim_deprecateCompressEncoding:4]: whoops yeah its deprecation 😆 

what do you mean by migration steps? are the customers not using compression already?

Comment (rnishtala-sumo) [reply .chloggen/chan-tim_deprecateCompressEncoding:4]: Yeah, this will be a breaking change.

Comment (chan-tim-sumo) [reply .chloggen/chan-tim_deprecateCompressEncoding]: added migration steps in a separate PR: https://github.com/SumoLogic/sumologic-otel-collector/pull/1605

File: exporter/sumologicexporter/config.go
Comment (sumo-drosiek) [inline exporter/sumologicexporter/config.go:98]: I was thinking about migration and our previous approach to breaking changes and we should do it similar way like https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/sumologicexporter/config.go#L110-L111.

So instead of getting error that `compress_encoding` is not a correct configuration, we should return more infotmative message with details what should be done to apply the valid configuration.

I do not expect from the customers to read changelog, and it is better to repeat the message in the logs, which they definitely read when the collector is crashing (due to invalid configuration).

Following the above example there is difference between the following information:

```
support for the graphite metric format was removed, please use prometheus or otlp instead
```

and

```
unexpected metric format: graphite
```

Comment (chan-tim-sumo) [reply exporter/sumologicexporter/config.go:98]: oh yeah this makes alot of sense, just to make sure i'm getting this right:

keep the `compress_encoding` field in the config struct, do something similar [like this](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/exporter/sumologicexporter/config.go#L110-L111) and throw an better detailed error message saying something like `support for compress encoding has been deprecated, please use compression instead`

Comment (sumo-drosiek) [reply exporter/sumologicexporter/config.go:98]: Yup, thats the conception

I would change `deprecated` to `removed`

File: exporter/sumologicexporter/config.go
Comment (sumo-drosiek) [inline exporter/sumologicexporter/config.go]: I would only check if `cfg.CompressEncoding` exist

Comment (sumo-drosiek) [reply exporter/sumologicexporter/config.go]: and if yes, I would raise an error that the option is deprecated

Comment (chan-tim-sumo) [reply exporter/sumologicexporter/config.go]: ohh 😆 done!

File: exporter/sumologicexporter/config.go
Comment (sumo-drosiek) [inline exporter/sumologicexporter/config.go:78]:
```suggestion
		if cfg.CompressEncoding != nil {
		return errors.New("support for `compress_encoding` configuration has been removeed in favor of `compression`")
```

File: exporter/sumologicexporter/config.go
Comment (sumo-drosiek) [inline exporter/sumologicexporter/config.go:31]:
```suggestion
	CompressEncoding *configcompression.Type `mapstructure:"compress_encoding"`
```

**Review by sumo-drosiek — COMMENTED** [review]
I tested the binary, and it seems that those changes are required, otherwise setting empty `compress_encoding` (which is valid for deprecated value) do not raise an error

File: exporter/sumologicexporter/config.go
Comment (chan-tim-sumo) [reply exporter/sumologicexporter/config.go]: > I tested the binary, and it seems that those changes are required, otherwise setting empty `compress_encoding` (which is valid for deprecated value) do not raise an error

good catch, I assumed`fmt.Errorf` would've raise an error, I guess not 😆

**Comment by chan-tim-sumo** [comment]
hey @dashpole , saw that you're one of the approvers. can you take a look at this pr? thanks!

**Comment by dashpole** [comment]
I'll take a look after it has been approved by a codeowner.

**Comment by rnishtala-sumo** [comment]
LGTM! :rocket:

**Review by rnishtala-sumo — APPROVED** [review]

**Review by dashpole — APPROVED** [review]

**Comment by chan-tim-sumo** [comment]
@dmitryax @TylerHelmuth need a maintainer to merge this, can you take a look? 💪

**Review by dmitryax — APPROVED** [review]

**Pull request merged**
